code/go/0chain.net/chaincore/httpclientutil/http_client_util_test.go

```diff
@@ -0,0 +1,1030 @@
+			},
+			want:    make([]byte, 0, 512),
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			got, err := SendPostRequest(tt.args.url, tt.args.data, tt.args.ID, tt.args.pkey, tt.args.wg)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("SendPostRequest() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("SendPostRequest() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSendMultiPostRequest(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		urls []string
+		data []byte
+		ID   string
+		pkey string
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "OK",
+			args: args{
+				urls: []string{
+					getTestServerURL(),
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			SendMultiPostRequest(tt.args.urls, tt.args.data, tt.args.ID, tt.args.pkey)
+		})
+	}
+}
+
+func TestSendTransaction(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		txn  *Transaction
+		urls []string
+		ID   string
+		pkey string
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "OK",
+			args: args{
+				urls: []string{
+					getTestServerURL(),
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			SendTransaction(tt.args.txn, tt.args.urls, tt.args.ID, tt.args.pkey)
+		})
+	}
+}
+
+func TestMakeGetRequest(t *testing.T) {
+	server := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				data := map[string]interface{}{
+					"key": "value",
+				}
+				blob, err := json.Marshal(data)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := rw.Write(blob); err != nil {
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer server.Close()
+
+	errServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				rw.WriteHeader(http.StatusInternalServerError)
+			},
+		),
+	)
+	defer errServer.Close()
+
+	type args struct {
+		remoteUrl string
+		result    interface{}
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Request_Creating_ERR",
+			args: args{
+				remoteUrl: string(rune(0x7f)),
+			},
+			wantErr: true,
+		},
+		{
+			name: "Client_ERR",
+			args: args{
+				remoteUrl: "/",
+			},
+			wantErr: true,
+		},
+		{
+			name: "Resp_Status_Not_Ok_ERR",
+			args: args{
+				remoteUrl: errServer.URL,
+			},
+			wantErr: true,
+		},
+		{
+			name: "JSON_Decoding_ERR",
+			args: args{
+				remoteUrl: server.URL,
+				result:    "}{",
+			},
+			wantErr: true,
+		},
+		{
+			name: "OK",
+			args: args{
+				remoteUrl: server.URL,
+				result:    &map[string]interface{}{},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Closure error, missing `tt:=tt`.

Reply:
There is no race between goroutines. Test running order is determined by `tests` struct slice.

Reply:
Should there and in the next mentioned cases be added parallel running?

Reply:
Yes they should all be run in parallel. These are methods are run as part of http requests, 0chain has to be able handle multiple simulanious http request. So these functions have to prove they can run in parallel. If this causes problems then the problmes need to be sovled.

The time `not` to run tests in parallel is when you are testing functions like `SetDebugLevel` that only ever get called once per execuation.